pkg/util/mount/mount.go

```diff
@@ -76,3 +80,51 @@ func GetMountRefs(mounter Interface, mountPath string) ([]string, error) {
 	}
 	return refs, nil
 }
+
+// GetDevicePrefixRefCount: given a prefix of device path, find its reference count from /proc/mounts
+// returns the reference count to the device and error code
+// for services like iscsi construct multiple device paths with the same prefix pattern.
+// this function aggregates all references to a service based on the prefix pattern
+func GetDevicePrefixRefCount(mounter Interface, deviceNamePrefix string) (int, error) {
+	mps, err := mounter.List()
+	if err != nil {
+		return -1, err
+	}
+
+	// Find the number of references to the device.
+	refCount := 0
+	for i := range mps {
+		if strings.HasPrefix(mps[i].Device, deviceNamePrefix) {
```

Comment:
This seems like a really niche function - better to just put it close to the one callsite.